src/DI/ContainerBuilder.php

```diff
@@ -463,9 +468,14 @@ private function resolveEntityClass($entity, $recursive = [])
 				throw new ServiceCreationException(sprintf("Factory '%s' used in service '%s' is not callable.", Nette\Utils\Callback::toString($entity), $name[0]));
 			}
 
-			$class = preg_replace('#[|\s].*#', '', PhpReflection::parseAnnotation($reflection, 'return'));
-			if ($class) {
-				$class = $refClass ? PhpReflection::expandClassName($class, $refClass) : ltrim($class, '\\');
+			if (PHP_VERSION_ID >= 70000 && $reflection->hasReturnType()) {
+				$type = $reflection->getReturnType();
+				$class = !$type->isBuiltIn() ? (string) $type : null;
```

Comment:
Well, `class: string` fails anyways on [other exception](https://github.com/nette/di/blob/master/src/DI/ContainerBuilder.php#L481).

Also, I found out about some problematic cases with the `@return` expanding (like, `@return string` in namespace becomes `Foo\string` - not a problem in 5.x, but in 7.0), so the exception in `resolveEntityClass()` is probably a better solution.